driver/src/http.rs

```diff
@@ -75,22 +92,36 @@ impl HttpClient {
     }
 
     /// Standard HTTP GET request that parses the result as JSON.
-    pub fn get_json<U, T>(&self, url: U) -> Result<T>
+    pub fn get_json<U, T>(&self, url: U, label: HttpLabel) -> Result<T>
     where
         Uri: TryFrom<U>,
         <Uri as TryFrom<U>>::Error: Into<HttpError>,
         T: DeserializeOwned,
     {
-        Ok(self.inner.get(url)?.json()?)
+        let start = Instant::now();
+
+        let json = self.inner.get(url)?.text()?;
+        let size = json.len();
+        let result = serde_json::from_str(&json)?;
+
+        self.metrics.request(label, start.elapsed(), size);
```

Comment:
ubernit: here you are technically also timing the deserialization (negligible) but we could also just move it above the `let result ...` line.

Reply:
Good point! Didn't think of that.